tests/TestFCI.py

```diff
@@ -1,95 +1,187 @@
+    "tests/TestData/benchmark_returned_results/bnlearn_discrete_10000_water_fci_chisq_0.05.txt": "48eee804d59526187b7ecd0519556ee5",
+    "tests/TestData/benchmark_returned_results/bnlearn_discrete_10000_hailfinder_fci_chisq_0.05.txt": "6b9a6b95b6474f8530e85c022f4e749c",
+    "tests/TestData/benchmark_returned_results/bnlearn_discrete_10000_hepar2_fci_chisq_0.05.txt": "e9de65c752011d72c36589b68590011d",
+    "tests/TestData/benchmark_returned_results/bnlearn_discrete_10000_win95pts_fci_chisq_0.05.txt": "648fdf271e1440c06ca2b31b55ef1f3f",
+    "tests/TestData/benchmark_returned_results/bnlearn_discrete_10000_andes_fci_chisq_0.05.txt": "939375acd3c623334b20c709bf8f347c",
+    "tests/TestData/benchmark_returned_results/linear_10_fci_fisherz_0.05.txt": "289c86f9c665bf82bbcc4c9e1dcec3e7"
+}
+
+INCONSISTENT_RESULT_GRAPH_ERRMSG = "Returned graph is inconsistent with the benchmark. Please check your code with the commit fb092d1."
+INCONSISTENT_RESULT_GRAPH_WITH_PAG_ERRMSG = "Returned graph is inconsistent with the truth PAG."
+
+# verify files integrity first
+for file_path, expected_MD5 in BENCHMARK_TXTFILE_TO_MD5.items():
+    with open(file_path, 'rb') as fin:
+        assert hashlib.md5(fin.read()).hexdigest() == expected_MD5,\
+            f'{file_path} is corrupted. Please download it again from https://github.com/cmu-phil/causal-learn/blob/fb092d1/tests/TestData'
+
 
 def gen_coef():
     return np.random.uniform(1, 3)
 
 
 class TestFCI(unittest.TestCase):
-
     def test_simple_test(self):
-        np.random.seed(0)
-        sample_size, loc, scale = 200, 0.0, 1.0
-        X1 = np.random.normal(loc=loc, scale=scale, size=sample_size)
-        X2 = X1 * gen_coef() + np.random.normal(loc=loc, scale=scale, size=sample_size)
-        X3 = X1 * gen_coef() + np.random.normal(loc=loc, scale=scale, size=sample_size)
-        X4 = X2 * gen_coef() + X3 * gen_coef() + np.random.normal(loc=loc, scale=scale, size=sample_size)
-        data = np.array([X1, X2, X3, X4]).T
-        G, edges = fci(data, fisherz, 0.05, verbose=False)
-        # pdy = GraphUtils.to_pydot(G)
-        # pdy.write_png('simple_test.png')
+        data = np.empty(shape=(0, 4))
+        true_dag = DiGraph()
+        true_dag.add_edges_from([(0, 1), (0, 2), (1, 3), (2, 3)])
+        G, edges = fci(data, d_separation, 0.05, verbose=False, true_dag=true_dag)
+
+        ground_truth_nodes = []
+        for i in range(4):
+            ground_truth_nodes.append(GraphNode(f'X{i+1}'))
+        ground_truth_dag = Dag(ground_truth_nodes)
+        ground_truth_dag.add_directed_edge(ground_truth_nodes[0], ground_truth_nodes[1])
+        ground_truth_dag.add_directed_edge(ground_truth_nodes[0], ground_truth_nodes[2])
+        ground_truth_dag.add_directed_edge(ground_truth_nodes[1], ground_truth_nodes[3])
+        ground_truth_dag.add_directed_edge(ground_truth_nodes[2], ground_truth_nodes[3])
```

Comment:
please do a for loop here, instead of manually add every edge.

e.g.

```
for node_1, node2 in [(0, 1), (0, 2), (1, 3), (2, 3)]:
    ground_truth_dag.add_directed_edge(ground_truth_nodes[node_1], ground_truth_nodes[node_1])
```